presto-orc/src/main/java/com/facebook/presto/orc/reader/SliceDirectStreamReader.java

```diff
@@ -192,7 +197,10 @@ public Block readBlock(Type type)
             dataStream.next(data, offset, offset + currentLength);
 
             // adjust offsetVector with truncated length
-            int truncatedLength = computeTruncatedLength(slice, offset, currentLength, type);
+            int truncatedLength = currentLength;
+            if (maxCodePointCount >= 0) {
```

Comment:
Maybe make this `maxCodePointCount >= 0 && currentLength > maxCodePointCount`?  I'm thinking it might help types that have a max length but the code never sees multibyte data.

Reply:
good point